lib/Backend/arm64/EncoderMD.cpp

```diff
@@ -1472,7 +1472,13 @@ bool EncoderMD::TryConstFold(IR::Instr *instr, IR::RegOpnd *regOpnd)
             return false;
         }
 
-        instr->ReplaceSrc(regOpnd, regOpnd->m_sym->GetConstOpnd());
+        IR::Opnd* constOpnd = regOpnd->m_sym->GetConstOpnd();
+        if (constOpnd->GetSize() == 8 && regOpnd->GetSize() < 8)
```

Comment:
>constOpnd->GetSize() == 8 && regOpnd->GetSize() [](start = 12, length = 47)

Should this check be

constOpnd->GetSize() > regOpnd->GetSize()

Reply:
I use == 8 && < 8 because regs of size 1 or 2 get converted to size 4 but they have already been converted by this point so yeah, I think your way is better.